### DISCUSSION TITLE
How to escape {{...}} ?

### DISCUSSION THREAD

**Discussion by gregorylevilain** [usebruno/bruno#1410] [Q&A]
I have a query body that includes a "{{fileName}}" which should remain as is. How can I escape it and avoid variable replacement ? 
Thanks a lot for your help, and for Bruno :)

**Comment by helloanoop** [comment]
This needs to be fixed. We use handlebars for interpolation, and we should allow escaping

This is the line: https://github.com/usebruno/bruno/blob/main/packages/bruno-electron/src/ipc/network/interpolate-string.js#L37

`const template = Handlebars.compile(str, { noEscape: true });`

@Its-treason @martinsefcik @mirkogolze I believe it should be safe to allow escaping. and remove the `{noEscape: true }`

**Reply by martinsefcik** [reply]
@helloanoop I don't think setting `noEscape: false` helps as it is for something different. If your variable contains text which is HTML then it is replaced in template as escaped/not escaped HTML based on this parameter.  https://handlebarsjs.com/guide/#html-escaping
So I think `noEscape` must stay `true` in Bruno.

I expect @gregorylevilain needs something different. He has `{{fileName}}` in request body and he wants to send it as `{{fileName}}` without any replacement applied. Currently it is replaced with empty string in case `fileName` variable is not defined.

I tried to register the following helper in Handlebars:
```javascript
Handlebars.registerHelper('helperMissing', (options) => options.fn(this));
```
based on the documentation https://handlebarsjs.com/guide/hooks.html#helpermissing and some googling

and now if you have in request body `{{fileName}}` and `fileName` variable is not defined then it is not touched and printed as `{{fileName}}`. 
![image](https://github.com/usebruno/bruno/assets/1073299/b0165c1d-2581-4b32-a68c-1cf12c06887e)

If `fileName` variable is defined then it is replaced with variable value.
![image](https://github.com/usebruno/bruno/assets/1073299/4ed54e1e-7b71-4567-88e6-7665176143bb)

I think the same behavior is in Postman.

**Reply by Its-treason** [reply]
Sounds like a good solution and would close #1056
